api/PclusterApiHandler.py

```diff
@@ -685,12 +691,76 @@ def _get_params(_request):
     params.pop("path")
     return params
 
+def check_tags():
+    try:
+        cost_explorer = boto3.client('ce')
+        inactive_tags = cost_explorer.list_cost_allocation_tags(Status='Inactive',TagKeys=PCLUSTER_COST_TAGS)
+        return {'TagStatus': not (len(inactive_tags['CostAllocationTags']) > 0)}
+    except Exception as e:
+        return {"message": str(e)}, 500
 
-# Proxy
+def activate_tags():
+    try:
+        cost_explorer = boto3.client('ce')
+        newlist = [{'TagKey': tag, 'Status': 'Active'} for tag in PCLUSTER_COST_TAGS]
+        cost_explorer.update_cost_allocation_tags_status(CostAllocationTagsStatus=newlist)
+        return {"message": "Activated Tags"}
+    except Exception as e:
+        return {"message": str(e)}, 500
+ 
+def get_graph_data():
+    try:
+        cluster_name = request.args.get("cluster_name")
```

Comment:
nit: I'd move these 3 assignments and the check / validation above the try as they shouldn't throw and it decreases the amount of the function that is wrapped in the try (which makes it easier to read as the checks are at the top).

Reply:
resolved